checker/src/org/checkerframework/checker/lock/LockStore.java

```diff
@@ -22,16 +27,23 @@
  */
 public class LockStore extends CFAbstractStore<CFValue, LockStore> {
 
+    /** If true, indicates that the store refers to a point in the code
+      * inside a constructor or initializer. This is useful because
```

Comment:
`inConstructorOrInitializer` is also set to true for static initializer blocks. However, in `LockStore.getValue`, the class name is not automatically considered as being held within a static initializer. Within a constructor or instance initializer, `this` is automatically considered as being held. For example:

    class Example {
      static class MyClass { Object field; }
      static final @GuardedBy("Example.class") MyClass s = new MyClass();

      static {
	  // Example.class is *not* automatically considered synchronized
	    s.field.toString(); // error: [contracts.precondition.not.satisfied.field] unguarded access to field, variable or parameter 's' guarded by 'Example'
	    synchronized (Example.class) {
	      s.field.toString(); // OK
	    }
      }

      @GuardedBy("this") MyClass m = new MyClass();

      Example() {
	  // 'this' is automatically considered synchronized
	    m.field.toString(); // OK
	    synchronized (this) {
	      m.field.toString(); // OK
	    }
      }
    }